### PR TITLE
specs, explicit body to avoid spread

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
To avoid these model getting spread, after the "spread is spread" feature.

We should already had `@body` and `@bodyRoot` in test, for explicit body (non-spread).
I will add a `@bodyIgnore` case in a follow-up PR.

# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp:100]: Any ask on the naming of the request body property?

As `body`, there will be a name change in Java (but it does not break API)
```diff
-    public PollerFlux<PollOperationDetails, GenerationResult> beginLongRunningRpc(GenerationOptions generationOptions) {
+    public PollerFlux<PollOperationDetails, GenerationResult> beginLongRunningRpc(GenerationOptions body) {
```

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp:100]: Here the model is all body properties, so `@body` and `@bodyRoot` should be equivalent.

Comment (tadelesh) [reply packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp]: in tcgc logic, if the body model has name, then use that name, else use `body`.

**Comment by tadelesh** [comment]
please do not merge after we have the final decision to do this approach of changing spec.

File: packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp
Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp]: Note that Java prefer e.g. `requestContext` instead of `body`.

File: packages/cadl-ranch-specs/http/parameters/spread/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/parameters/spread/main.tsp:28]: I've changed the scenarioDoc, as now `op spreadAsRequestBody(...BodyParameter): NoContentResponse;` would be a spread.

**Review by msyyc — APPROVED** [review]
LGTM

**Review by tadelesh — COMMENTED** [review]
do you think we need to follow the patten from https://github.com/Azure/azure-rest-api-specs/pull/29483/files, use the template as follows?
```
alias BodyParameter<
  T,
  TName extends valueof string = "body",
  TDoc extends valueof string = ""
> = {
  @doc(TDoc)
  @friendlyName(TName)
  @bodyRoot
  body: T;
};
```

**Comment by weidongxu-microsoft** [comment]
> do you think we need to follow the patten from https://github.com/Azure/azure-rest-api-specs/pull/29483/files, use the template as follows?
> 
> ```
> alias BodyParameter<
>   T,
>   TName extends valueof string = "body",
>   TDoc extends valueof string = ""
> > = {
>   @doc(TDoc)
>   @friendlyName(TName)
>   @bodyRoot
>   body: T;
> };
> ```

It is test and most of them is valid for generic/unbranded. So far, I do not expect unbranded would write this `BodyParameter`, therefore it seems no reason to write this in cadl-ranch.

For the few "azure" tests, I would use that when `BodyParameter` gets to Azure.Core.

File: packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp:109]: switch to use local `BodyParameter` in azure.

will switch to `Azure.Core.BodyParameter` in future.

**Review by tadelesh — APPROVED** [review]

File: packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/core/lro/rpc/main.tsp:109]: `@doc` default as `TDoc`

**Pull request merged**
